### PR TITLE
Fix builds on non glibc based linux platforms

### LINKED ISSUE #477: Build fail on Alpine linux
### Describe the bug

A description of what the bug is.

### Steps to reproduce the behavior
1. Building master on Alpine linux
2. View output
3. See error
```
localhost:~/shared/work/aeron/msquic/build# ninja
[1/27] Building C object src/platform/CMakeFiles/platform.dir/datapath_linux.c.o
FAILED: src/platform/CMakeFiles/platform.dir/datapath_linux.c.o
/usr/bin/cc  -I../src/inc -Iopenssl/include -DQUIC_PLATFORM_LINUX -fms-extensions -fPIC -pthread -Wl,--no-as-needed -ldl -DQUIC_EVENTS_LTTNG -DQUIC_LOGS_STUB -DQUIC_DISABLE_RESUMPTION -DQUIC_DISABLE_0RTT_TESTS   -Werror -Wall -Wextra -Wformat=2 -Wno-type-limits -Wno-unknown-pragmas -Wno-unused-value -MD -MT src/platform/CMakeFiles/platform.dir/datapath_linux.c.o -MF src/platform/CMakeFiles/platform.dir/datapath_linux.c.o.d -o src/platform/CMakeFiles/platform.dir/datapath_linux.c.o   -c ../src/platform/datapath_linux.c
../src/platform/datapath_linux.c: In function 'QuicDataPathWorkerThread':
../src/platform/datapath_linux.c:2187:13: error: implicit declaration of function 'TEMP_FAILURE_RETRY' [-Werror=implicit-function-declaration]
 2187 |             TEMP_FAILURE_RETRY(
      |             ^~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
[2/27] Building C object src/platform/CMakeFiles/platform.dir/platform_linux.c.o
FAILED: src/platform/CMakeFiles/platform.dir/platform_linux.c.o
/usr/bin/cc  -I../src/inc -Iopenssl/include -DQUIC_PLATFORM_LINUX -fms-extensions -fPIC -pthread -Wl,--no-as-needed -ldl -DQUIC_EVENTS_LTTNG -DQUIC_LOGS_STUB -DQUIC_DISABLE_RESUMPTION -DQUIC_DISABLE_0RTT_TESTS   -Werror -Wall -Wextra -Wformat=2 -Wno-type-limits -Wno-unknown-pragmas -Wno-unused-value -MD -MT src/platform/CMakeFiles/platform.dir/platform_linux.c.o -MF src/platform/CMakeFiles/platform.dir/platform_linux.c.o.d -o src/platform/CMakeFiles/platform.dir/platform_linux.c.o   -c ../src/platform/platform_linux.c
../src/platform/platform_linux.c: In function 'QuicThreadCreate':
../src/platform/platform_linux.c:647:18: error: implicit declaration of function 'pthread_attr_setaffinity_np'; did you mean 'pthread_setaffinity_np'? [-Werror=implicit-function-declaration]
  647 |             if (!pthread_attr_setaffinity_np(&Attr, sizeof(CpuSet), &CpuSet)) {
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                  pthread_setaffinity_np
cc1: all warnings being treated as errors
[12/27] Building CXX object src/test/bin/CMakeFiles/msquictest.dir/quic_gtest.cpp.o
ninja: build stopped: subcommand failed.
```

### Expected vs actual behavior

A description of what you expected to happen and what actually happened.

### PR DESCRIPTION
**Pull request by ThadHouse**
TEMP_FAILURE_RETRY is a glibc extension, but the spec is very well defined, and easy to recreate manually. 

pthread attribute setting for affinity doesn't exist, so only use the attribute if glibc is defined, otherwise set the thread affinity after creating the thread.

The logging disable change in CMake is needed because I couldn't figure out how to install LTTng on Alpine. 

Fixes #477

### CODE REVIEW

File: src/platform/datapath_linux.c [hunk 1]
```diff
@@ -2171,6 +2171,17 @@ QuicDataPathBindingGetLocalMtu(
 #endif
 }
 
+#ifndef TEMP_FAILURE_RETRY
+#define TEMP_FAILURE_RETRY(expression)                                  \
+        ({                                                              \
```
Comment (nibanks) [inline src/platform/datapath_linux.c:2176]: nit: only 4 spaces for indent please.

**Comment by ThadHouse** [comment]
It actually turns out alpine is kind of a pain to get running on Azure because Azure Pipelines itself depends on glibc for Node.JS. But this does get it building correctly now.

File: src/platform/datapath_linux.c [hunk 2]
```diff
@@ -2171,6 +2171,17 @@ QuicDataPathBindingGetLocalMtu(
 #endif
 }
 
+#ifndef TEMP_FAILURE_RETRY
+#define TEMP_FAILURE_RETRY(expression)                              \
+    ({                                                              \
```
Comment (nibanks) [inline src/platform/datapath_linux.c:2176]: Why does it have wrapping parentheses?

File: src/platform/platform_linux.c [hunk 3]
```diff
@@ -637,6 +637,7 @@ QuicThreadCreate(
         return errno;
     }
 
+#ifdef __GLIBC__  // TODO figure out
```
Comment (nibanks) [inline src/platform/platform_linux.c:640]: Do you still need this TODO comment?

File: src/platform/datapath_linux.c (hunk 2 above)
Comment (ThadHouse) [reply src/platform/datapath_linux.c:2176]: Just force of habit, usually wrap expressions in parenthesis in macros for safety.

Comment (nibanks) [reply src/platform/datapath_linux.c:2176]: They are unnecessary in this case. The curly braces already do everything.

Comment (ThadHouse) [reply src/platform/datapath_linux.c:2176]: Actually, it fails without them.

```
../../src/platform/datapath_linux.c: In function 'QuicDataPathWorkerThread':
../../src/platform/datapath_linux.c:2177:9: error: expected expression before 'long'
 2177 |         long int FailureRetryResult = 0;                            \
      |         ^~~~
../../src/platform/datapath_linux.c:2203:13: note: in expansion of macro 'TEMP_FAILURE_RETRY'
 2203 |             TEMP_FAILURE_RETRY(
      |             ^~~~~~~~~~~~~~~~~~
```

Comment (nibanks) [reply src/platform/datapath_linux.c:2176]: Ok. Weird. This code is doing something (getting a return value) from this macro that I didn't know was possible:
```
        int ReadyEventCount =
            TEMP_FAILURE_RETRY(
                epoll_wait(
                    ProcContext->EpollFd,
                    EpollEvents,
                    EpollEventCtMax,
                    -1));
```

**Pull request merged**